framework/src/vectorpostprocessors/ElementVariablesDifferenceMax.C

```diff
@@ -0,0 +1,123 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ElementVariablesDifferenceMax.h"
+
+// MOOSE includes
+#include "MooseVariable.h"
+
+// libMesh includes
+#include "libmesh/quadrature.h"
+
+template <>
+InputParameters
+validParams<ElementVariablesDifferenceMax>()
+{
+  InputParameters params = validParams<ElementVectorPostprocessor>();
+
+  params.addRequiredCoupledVar(
+      "compare_a",
+      "The first variable to evaluate the difference with, performed as \"compare_a - compare_b\"");
+  params.addRequiredCoupledVar("compare_b",
+                               "The second variable to evaluate the difference with, "
+                               "performed as \"compare_a - compare_b\"");
+
+  params.addParam<bool>(
+      "furthest_from_zero", false, "Find the difference with the highest absolute value");
+
+  return params;
+}
+
+ElementVariablesDifferenceMax::ElementVariablesDifferenceMax(const InputParameters & parameters)
+  : ElementVectorPostprocessor(parameters),
+    _a(coupledValue("compare_a")),
+    _b(coupledValue("compare_b")),
+    _a_value(declareVector(getVar("compare_a", 0)->name())),
+    _b_value(declareVector(getVar("compare_b", 0)->name())),
+    _max_difference(declareVector("Difference")),
+    _position_x(declareVector("X")),
+    _position_y(declareVector("Y")),
+    _position_z(declareVector("Z")),
+    _furthest_from_zero(getParam<bool>("furthest_from_zero"))
+{
+  // These are all single-value arrays
+  _a_value.resize(1);
+  _b_value.resize(1);
+  _max_difference.resize(1);
+  _position_x.resize(1);
+  _position_y.resize(1);
+  _position_z.resize(1);
+}
+
+void
+ElementVariablesDifferenceMax::execute()
+{
+  for (unsigned int qp = 0; qp < _qrule->n_points(); ++qp)
+  {
+    // Get the difference
+    const Real difference = _furthest_from_zero ? std::abs(_a[qp] - _b[qp]) : _a[qp] - _b[qp];
```

Comment:
If this is regarding the failing tests in parallel, I just looked into the cause. The differences are only in the location columns and not the value columns. I believe the issue is with the symmetry of the test problem: the same `_max_difference` exists in multiple locations. The symmetry even means that the values for `_a` and `_b` are the same but the locations are different. Different mesh partitions in parallel means that a different location may be found.

In the actual use case the user will _hopefully_ be aware of the simulation symmetry and use this information to identify the region that contains the maximum differences.

I will update the tests to eliminate the symmetry.